main.py

```diff
@@ -99,13 +100,29 @@ async def on_message(message: Message):
 
 # Take a filename as string and return it formatted nicely
 def format_filename(filename: str):
-    # Remove file extension
-    filename = path.splitext(filename)[0]
 
-    # Replace all underscores with spaces
-    filename = filename.replace("_", " ")
+    # Get all the tags for a track
+    audio = TinyTag.get(
+        str(os.path.join(f"{attachment_directory_filepath}", f"{filename}"))
+    )
+    content = ""
+    # If the tag doesnot exist or is whitespace display the file name only
+    # Otherwise display in the format @user: <Artist-tag> - <Title-tag>
+    if audio.artist is not None and len(audio.artist.strip()) != 0:
+        content = content + f"{str(audio.artist)} - "
+
+    if audio.title is not None:
+        if len(audio.title.strip()) == 0:
+            filename = path.splitext(filename)[0]
+            content = content + filename.replace("_", " ")
+        else:
+            content = content + f"{str(audio.title)}"
+    # If the title tag doesnot exist but the artist tag exists, display the file name along with artist tag
+    else:
+        filename = path.splitext(filename)[0]
+        content = content + filename.replace("_", " ")
```

Comment:
We can remove some duplicated code if we write the if statements like this
```suggestion
    if audio.title is not None and not audio.title.isspace():
        content = content + f"{str(audio.title)}"
    # If the title tag does not exist but the artist tag exists, display the file name along with artist tag
    else:
        filename = path.splitext(filename)[0]
        content = content + filename.replace("_", " ")
```